protocol/messenger_communities.go

```diff
@@ -2305,21 +2305,44 @@ func (m *Messenger) ShareCommunity(request *requests.ShareCommunity) (*Messenger
 	if err := request.Validate(); err != nil {
 		return nil, err
 	}
+
+	var messageStatusLinkPreview common.StatusLinkPreview
 	response := &MessengerResponse{}
 
 	community, err := m.communitiesManager.GetByID(request.CommunityID)
 	if err != nil {
 		return nil, err
 	}
 
+	communityURL, err := m.ShareCommunityURLWithChatKey(community.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	communityLinkPreview := &common.StatusCommunityLinkPreview{
+		CommunityID:  community.IDString(),
+		DisplayName:  community.Name(),
+		Description:  community.DescriptionText(),
+		MembersCount: uint32(community.MembersCount()),
+		Color:        community.Color(),
+	}
+
+	messageStatusLinkPreview.Community = communityLinkPreview
+	messageStatusLinkPreview.URL = communityURL
+
 	var messages []*common.Message
 	for _, pk := range request.Users {
 		message := common.NewMessage()
+		message.StatusLinkPreviews = []common.StatusLinkPreview{messageStatusLinkPreview}
 		message.ChatId = pk.String()
-		message.CommunityID = request.CommunityID.String()
 		message.Shard = community.Shard().Protobuffer()
-		message.Text = fmt.Sprintf("Community %s has been shared with you", community.Name())
+		message.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+		if err != nil {
+			return nil, err
+		}
+		message.Text = communityURL
 		if request.InviteMessage != "" {
+			fmt.Sprintf("%s\n%s", request.InviteMessage, communityURL)
 			message.Text = request.InviteMessage
```

Comment:
Shouldn't it be a direct assignment?
```suggestion
			message.Text = fmt.Sprintf("%s\n%s", request.InviteMessage, communityURL)
```